app/reducers/ui.js

```diff
@@ -5,19 +5,76 @@ import initialAppState from './initialAppState';
 
 const initialState = initialAppState.ui;
 
+const addWindowtoUI = ( state, payload ) =>
+{
+    const targetWindowId = payload.windowId;
+    const getCurrentWindowState = state.windows;
+    const windowState = [ ...getCurrentWindowState ];
+    windowState.push( { windowId: targetWindowId, settingsMenuIsVisible: false } );
+    const newState = { ...state, windows: windowState };
+    return newState;
+};
+function ToggleMenu( state, payload, showMenu )
+{
+    const targetWindowId = payload.windowId;
+    const getCurrentWindowState = state.windows;
+    const windowState = [ ...getCurrentWindowState ];
+    const found = windowState.find( (obj) => obj.windowId === targetWindowId );
+    if ( !found )
+    {
+        return state;
+    }
+    const updatedWindowIndex = windowState.findIndex( (obj) => obj.windowId === targetWindowId );
+    const updatedWindow = { ...found };
+    if ( showMenu === true )
+    {
+        updatedWindow.settingsMenuIsVisible = true;
+    }
+    else
+    {
+        updatedWindow.settingsMenuIsVisible = false;
+    }
+    windowState[updatedWindowIndex] = updatedWindow;
+    const newState = { ...state, windows: windowState };
+    return newState;
+}
+
+const showSettingsMenu = ( state, payload ) =>
```

Comment:
these two funcs, show/hide could be removed and the func called directly from the reducer.